pyroSAR/snap/auxil.py

```diff
@@ -274,22 +275,169 @@ def __init__(self):
                     'You can define a different one with ExamineSnap.set_path()'
                     ''.format(options='\n'.join(executables), select=self.path), UserWarning)
 
-            self.__get_etc()
             self.__read_config()
+            self.__read_snap_properties()
+            self.__set_properties_paths()
+
+    def __read_config(self):
+        """
+        This method reads the config.ini to examine the snap paths. If the snap paths are not in the config.ini or the
+        paths are wrong they will be automatically created.
+
+        Returns
+        -------
+        None
+
+        """
+        if 'SNAP' in ConfigHandler.sections:
+            if 'etc' in ConfigHandler.keys('SNAP'):
+                self.etc = ConfigHandler.get('SNAP', 'etc')
+
+                if not os.path.exists(self.etc):
+                    self.__get_etc()
+
+            else:
+                self.__get_etc()
+
+            if 'properties' in ConfigHandler.keys('SNAP'):
+                self.properties = ConfigHandler.get('SNAP', 'properties')
+
+                if not os.path.exists(self.properties):
+                    self.__get_properties()
+
+            else:
+                self.__get_properties()
+
+            if 'auxdatapath' in ConfigHandler.keys('SNAP'):
```

Comment:
The auxdatapath handling might be easier by using property and setter methods:
```Python
    @property
    def auxdatapath(self):
        if not hasattr(self, '__auxdatapath'):
            path_default = os.path.join(os.path.expanduser('~'), '.snap', 'auxdata')
            if 'auxdatapath' in ConfigHandler.keys('SNAP'):
                path_conf = ConfigHandler.get('SNAP', 'auxdatapath')
                # if the path in config is a directory, set this path to the private 
                # __auxdatapath attribute; if not, use the auxdata setter to directly 
                # write the default directory to the config
                if os.path.exists(path_conf):
                    self.__auxdatapath = path_conf
                else:
                    self.auxdatapath = path_default
            else:
                self.auxdatapath = path_default
        return self.__auxdatapath
    
    @auxdatapath.setter
    def auxdatapath(self, path):
        self.__auxdatapath = path
        ConfigHandler.set('SNAP', 'auxdatapath', path, True)

```

Reply:
Thank you very much for your efforts. I will change that.